### PR TITLE
Don't specialize torch.Size with specialize_int = False

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96419

Fixes https://github.com/pytorch/pytorch/issues/95868

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: test/dynamo/test_unspec.py [hunk 1]
```diff
@@ -199,6 +199,19 @@ def fn(x, y):
             res = opt_fn(x, y)
             self.assertTrue(same(ref, res))
 
+    def test_shape_graph_break(self):
+        from torch._dynamo.comptime import comptime
+
+        def fn(x):
+            x_shape = x.size()
+            comptime.graph_break()
```
Comment (ngimel) [inline test/dynamo/test_unspec.py:207]: should we use `comptime.graph_break()` everywhere now, instead of `torch._dynamo.graph_break()` ?

**Review by ngimel — APPROVED** [review]

File: test/dynamo/test_unspec.py (hunk 1 above)
Comment (ezyang) [reply test/dynamo/test_unspec.py:207]: uhh, it doesn't really matter. If people like, I can prefer `torch._dynamo.graph_break` and delete the comptime version

**Pull request closed without merging**